### DISCUSSION TITLE
How to display the attributes of an entity from an automation

### DISCUSSION THREAD

**Discussion by rbr101** [Blackymas/NSPanel_HA_Blueprint#813] [Q&A] [answered]
Hi,

I want to display the attributes for an entity from an homeassistant/node red automation. So for example when i switch on the light, the dimmer/color attributes of that light should be displayed on the screen for a period. Is this possible with the currrent implementation?

I'm switching from the JoBro99 implementation and there i can do this from an automation using a service. (ESPHome: nspanel_woonkamer_show_entity)

I use this feature a lot so i don't have easy access to the attributes when switched on.

**Comment by edwardtfn** [comment]
I don't know about the implementation you mentioned, however I can say this implementation here also allows setting the fields in the screen by using automations and calling services.This is exactly what the blueprint is doing.
But probably would be easier if you just create template sensors with the attributes you wanna see in the panel and then assign that new entity to the panel, no?

**Comment by rbr101** [comment]
Thanks , I will figure out. Great work by the way!

My use case was that when motion detected the lights switch on, I want to display the dim/color on the display for that entity using an node red automation.

Just replaced one of the units, only had some difficulties uploading the TFT.  Had to flash Tasmota and use the berry driver to finally upload the nspanel_eu.tft, but besides that everything works great. Much better response when touching the display than the JoBro99 implementation. 

Seeing forward for the media_player support!

Cheers,
R

**Reply by edwardtfn** [reply, marked as answer]
What I'm writing here is not a supported way of using it, therefore you shouldn't expect the project's team supporting you with this, ok?

You can try using the light settings page by calling setting the `sensor.xxxxx_settings_entity` using the service `esphome.xxxxx_set_settings_entity`. It's value should be a json like string with `page` indicating the page initiating the call (in your case you probably can use `home` and `entity` should be the `entity_id` of the light you want to set.
Something like this:
```yaml
service: esphome.office_workstation_panel_set_settings_entity
data:
  entity: '{"page": "home", "entity": "light.office_ceiling_lines"}'
```

**Reply by rbr101** [reply]
Thanks,

That was exact what I was looking for, just tested and works perfect!

**Comment by doron1** [comment]
You may want to check out issue #802 for a possible solution.

**Reply by rbr101** [reply]
No it's not about attributes of an entity, I wan't to show the dim/color settings of the entity when a light is switches on
